CMakeLists.txt

```diff
@@ -188,28 +188,28 @@ target_include_directories(onnx PUBLIC ${ONNX_ROOT} "${CMAKE_CURRENT_BINARY_DIR}
 target_link_libraries(onnx PUBLIC onnx_proto)
 
 if(BUILD_PYTHON)
-  add_library(onnx_pybind MODULE "${ONNX_ROOT}/onnx/cpp2py_export.cc")
-  target_include_directories(onnx_pybind PRIVATE "${CMAKE_CURRENT_BINARY_DIR}" "${PROTOBUF_INCLUDE_DIRS}" "${PYTHON_INCLUDE_DIR}")
+  add_library(onnx_cpp2py_export SHARED "${ONNX_ROOT}/onnx/cpp2py_export.cc")
+  target_include_directories(onnx_cpp2py_export PRIVATE "${CMAKE_CURRENT_BINARY_DIR}" "${PROTOBUF_INCLUDE_DIRS}" "${PYTHON_INCLUDE_DIR}")
 
   # pybind11 is a header only lib
   find_package(pybind11)
   if(pybind11_FOUND)
-    target_include_directories(onnx_pybind PRIVATE ${pybind11_INCLUDE_DIRS})
+    target_include_directories(onnx_cpp2py_export PRIVATE ${pybind11_INCLUDE_DIRS})
   else()
-    target_include_directories(onnx_pybind PRIVATE ${ONNX_ROOT}/third_party/pybind11/include)
+    target_include_directories(onnx_cpp2py_export PRIVATE ${ONNX_ROOT}/third_party/pybind11/include)
   endif()
-  target_link_libraries(onnx_pybind PRIVATE onnx_proto onnx)
+  target_link_libraries(onnx_cpp2py_export PRIVATE "-Wl,--whole-archive" onnx_proto onnx "-Wl,--no-whole-archive")
```

Comment:
Depending on the compiler we might need different flags. This is `GCC` specific.  Check
https://github.com/caffe2/caffe2/blob/53cc7dfe10fa1edd5d731654bea21d317375ffbf/cmake/public/utils.cmake#L10-L24

Reply:
++ will do